### ISSUE TITLE
Allow user to request feedback when saving

### ISSUE THREAD

**Issue by bhousel** [openstreetmap/iD#4133]
This comes up for discussion sometimes in other channels but we don't have an issue to track it here.  

The idea would be to have a checkbox labelled something like "I'd like somebody to review these edits".  Then iD would add a special changeset tag (e.g. `review_requested=yes`?) and it would get special treatment by review systems like [OSMCha](https://osmcha.mapbox.com/).

similar: #3968

**Comment by willemarcel** [comment]
@bhousel It's a good idea! Some time ago someone suggested to add a hash tag like `#pleasereview` to the changeset comment, so we could track it on osmcha. Let me know if I can help on this!

I think we could add the information both in the changeset tag and in the comment. The tag makes it easier to be identified by software and the comment lets it more visible to users.

**Comment by bhousel** [comment]
> Some time ago someone suggested to add a hash tag like #pleasereview to the changeset comment, so we could track it on osmcha. Let me know if I can help on this!

Nice, I'll make sure that when this feature is implemented we will open an issue on OSMCha to support it.

Hashtags in comments are sort of falling out of favor, see #2834.  I think they are mostly used by the Missing Streets project for their dashboard.  There seems to be general agreement that people would rather see hashtags in a different changeset tag and encourage mappers to write real comments about what they are doing, but there will be some transition period where they still appear in comments.

**Comment by bhousel** [comment]
wip at #4223

**Comment by JamesKingdom** [comment]
Completed in https://github.com/openstreetmap/iD/commit/0ea043b31132815643a4cc6e0f7108bdf6de744f

**Comment by HolgerJeromin** [comment]
iD invented here a new tag. 
This tag should be IMO documented on the wiki, as this is designed to be used by other software.

**Comment by willemarcel** [comment]
@HolgerJeromin Thanks for your suggestion. I added it on https://wiki.openstreetmap.org/wiki/Changeset#Tags_on_changesets

**Comment by ltog** [comment]
I like the idea of giving new users the possibility to request a review. However, I am missing some key information:

1. What work has to be done to qualify as a review?
2. What are the possible outcomes of a review (maybe "good", "needs improvement", "needs revert"?) ?
3. Who is qualified to do a review?
4. How can we recognize if a review has been performed? Is it when (any kind of) a changeset comment has been given?
5. Am I overthinking this?

I'd like to see a list of software that evaluates `review_requested=yes` tags. @joto evaluates project files (see https://wiki.openstreetmap.org/wiki/Taginfo/Projects ) but AFAIK these are meant for normal tags only, not for changeset tags. Maybe software authors, which support this tag can mention this fact on the wiki?

**Comment by HolgerJeromin** [comment]
@ltog we have no life cycle of a review.
The problem of this is that many persons are reviewing the same with no way to handle situations like "changeset was not good, but all was fixed in later changesets. No need to look at it".

Perhaps osmcha has this internally, but this should work across other QA software tools...

**Comment by batpad** [comment]
> The problem of this is that many persons are reviewing the same with no way to handle situations like "changeset was not good, but all was fixed in later changesets. No need to look at it".

My two bits: this information should not live on external review tools but on the osm.org changeset page, which is the commonly understood canonical page for a changeset. I believe discussions on changesets accommodate these use cases just fine. I.e. if someone reviews a changeset, fixes things, they should leave a discussion on the changeset saying "changeset was not good, but all was fixed in later changesets. No need to look at it" for eg. All any external tool needs to do then is surface the current changeset discussions to the reviewer.

It would probably be a good idea for any external review tool to post back any internal taxonomy for tagging / rating changesets as changeset discussions so this information is surfaced in a common place on osm.org. I'm not sure if posting changeset discussions by bots is allowed / encouraged, but I'll create a ticket in OSMCha for this.

tl;dr: I think most of these use-cases are currently handled by the changeset comment tag and changeset discussions.

**Comment by HolgerJeromin** [comment]
@batpad Thanks for your input. 
You are right, the life cycle is (with our current software situation) best tracked in the changeset comments. 
I would propose to have further discussions about the topic in the linked osmcha tracker.
